File: TransCoder_tokenized_test_set_functions/cpp/SIEVE_ERATOSTHENES_0N_TIME_COMPLEXITY.cpp
void manipulated_seive ( int N ) {
  isprime [ 0 ] = isprime [ 1 ] = false;
  for ( long long int i = 2;
  i < N;
  i ++ ) {
    if ( isprime [ i ] ) {
      prime . push_back ( i );
      SPF [ i ] = i;
    }
    for ( long long int j = 0;
    j < ( int ) prime . size ( ) && i * prime [ j ] < N && prime [ j ] <= SPF [ i ];
    j ++ ) {
      isprime [ i * prime [ j ] ] = false;
      SPF [ i * prime [ j ] ] = prime [ j ];
    }
  }
}

File: TransCoder_tokenized_test_set_functions/java/SIEVE_ERATOSTHENES_0N_TIME_COMPLEXITY.java
static void manipulated_seive ( int N ) {
  isprime . set ( 0, false );
  isprime . set ( 1, false );
  for ( int i = 2;
  i < N;
  i ++ ) {
    if ( isprime . get ( i ) ) {
      prime . add ( i );
      SPF . set ( i, i );
    }
    for ( int j = 0;
    j < prime . size ( ) && i * prime . get ( j ) < N && prime . get ( j ) <= SPF . get ( i );
    j ++ ) {
      isprime . set ( i * prime . get ( j ), false );
      SPF . set ( i * prime . get ( j ), prime . get ( j ) );
    }
  }
}